geom_mgr/CubedSphere/CubedSphereGeomSpec_smod.F90

```diff
@@ -0,0 +1,290 @@
+      class default
+         equal_to = .false.
+      end select
+
+      contains
+      pure logical function equal_schmidt(a,b)
+         type(ESMF_CubedSphereTransform_Args), intent(in) :: a
+         type(ESMF_CubedSphereTransform_Args), intent(in) :: b
+
+         equal_schmidt = (a%target_lat /= b%target_lat) .and. &
+                        (a%target_lon /= b%target_lon) .and. &
+                        (a%stretch_factor /= b%stretch_factor)
+      end function equal_schmidt
+
+   end function equal_to
+
+
+   ! HConfig section
+   module function make_CubedSphereGeomSpec_from_hconfig(hconfig, rc) result(spec)
+      type(CubedSphereGeomSpec) :: spec
+      type(ESMF_HConfig), intent(in) :: hconfig
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+      logical :: found
+
+      spec%im_world = ESMF_HConfigAsI4(hconfig, keyString='im_world', asOkay=found, _RC)
+      _ASSERT(found, '"im_world" not found.') 
+      spec%decomposition = make_Decomposition(hconfig, cube_size=spec%im_world, _RC)
+      spec%schmidt_parameters = make_SchmidtParameters_from_hconfig(hconfig, _RC)
+
+      _RETURN(_SUCCESS)
+   end function make_CubedSphereGeomSpec_from_hconfig
+
+   function make_SchmidtParameters_from_hconfig(hconfig, rc) result(schmidt_parameters)
+      type(ESMF_CubedSphereTransform_Args) :: schmidt_parameters
+      type(ESMF_HConfig), intent(in) :: hconfig
+      integer, intent(out), optional :: rc
+
+      integer :: status, ifound
+      logical :: is_stretched, has_tlon, has_tlat, has_sfac, consistent
+  
+      ifound = 0 
+      has_sfac = ESMF_HConfigIsDefined(hconfig, keystring='stretch_factor', _RC)
+      if (has_sfac) then
+         schmidt_parameters%stretch_factor = ESMF_HConfigAsR8(hconfig, keystring='stretch_factor' ,_RC)
+         ifound = ifound + 1
+      end if
+      has_tlon = ESMF_HConfigIsDefined(hconfig, keystring='target_lon', _RC)
+      if (has_tlon) then
+         schmidt_parameters%target_lon = ESMF_HConfigAsR8(hconfig, keystring='target_lon' ,_RC)
+         schmidt_parameters%target_lon = schmidt_parameters%target_lon * MAPL_DEGREES_TO_RADIANS_R8
+         ifound = ifound + 1
+      end if
+      has_tlat = ESMF_HConfigIsDefined(hconfig, keystring='target_lat', _RC)
+      if (has_tlat) then
+         schmidt_parameters%target_lat = ESMF_HConfigAsR8(hconfig, keystring='target_lat' ,_RC)
+         schmidt_parameters%target_lat = schmidt_parameters%target_lat * MAPL_DEGREES_TO_RADIANS_R8
+         ifound = ifound + 1
+      end if
+      is_stretched = all([has_sfac, has_tlon, has_tlat])
+      consistent = (ifound .eq. 3) .or. (ifound .eq. 0)
+      _ASSERT(consistent, "specfied partial stretch parameters")
+      if (.not. is_stretched) then
+         schmidt_parameters%stretch_factor = undef_schmidt 
+         schmidt_parameters%target_lon= undef_schmidt 
+         schmidt_parameters%target_lat= undef_schmidt 
+      end if
```

Comment:
Not worth the change,  but as an exercise in software engineering choices...

 I suspect it would be a little bit clearer if the default (undef) block is moved to the top and just always performed.  (Inefficient, but ... it is just 3 assignments.   Then for each parameter you return if not found.   The 1st one without an error.  Subsequent ones return an error due to inconsistency.    Eliminates the `ifound = ifound + 1` and such.